src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -137,6 +144,13 @@ CHIP_ERROR DiscoveryImplPlatform::GenerateRotatingDeviceId(char rotatingDeviceId
 
 CHIP_ERROR DiscoveryImplPlatform::SetupHostname()
 {
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+    static char hostname[17]; // Hostname is 64-bit EUI-64 expressed as a 16-character hexadecimal string.
+    uint8_t eui64[8];
+    chip::DeviceLayer::ThreadStackMgr().GetFactoryAssignedEUI64(eui64);
+    snprintf(hostname, sizeof(hostname), "%02X%02X%02X%02X%02X%02X%02X%02X", eui64[7], eui64[6], eui64[5], eui64[4], eui64[3],
```

Comment:
yeah.. the spec doesn't specify it clearly if EUI64 should be formatted as little-endian 64-bit value or just a byte string. Not sure if it should be reversed.... @LuDuda what does your OT experience tell you? :)

Reply:
My OT experience :)? At least OpenThread CLI prints it from 0 to 7.

Reply:
That's right, I checked it using `ot eui64`. Fixed.